src/compiler/scala/tools/nsc/symtab/classfile/ClassfileParser.scala

```diff
@@ -674,9 +674,9 @@ abstract class ClassfileParser(reader: ReusableInstance[ReusableDataReader]) {
                           val tp = sig2type(tparams, skiptvs)
                           // sig2type seems to return AnyClass regardless of the situation:
                           // we don't want Any as a LOWER bound.
-                          if (tp.typeSymbol == AnyClass) TypeBounds.empty
-                          else TypeBounds.lower(tp)
-                        case '*' => TypeBounds.empty
+                          if (tp.typeSymbol == AnyClass) TypeBounds.upper(definitions.ObjectTpeJava)
```

Comment:
Can you add an example for the importance of distinguishing `AnyTpe` and `ObjectTpeJava` here?